### ISSUE TITLE
Caplin fails to sync from scratch

### ISSUE THREAD

**Issue by errge** [erigontech/erigon#13212]
#### System information

Erigon version: 3.00.0-alpha7-34714c0c

OS & Version: Linux

Commit hash: 34714c0c

Erigon Command (with flags/config):
```erigon --datadir ./data --prune.mode archive --http.api=eth,erigon,web3,net,debug,trace,txpool --caplin.archive --diagnostics.disabled --nat none --private.api.addr "" --beacon.api beacon,builder,config,debug,events,node,validator,lighthouse --beacon.api.port 3500 --http.port 8545 --ws --ws.port 8546 --authrpc.port 8551 --torrent.port 20202 --port 30303 --p2p.protocol 68 --p2p.allowed-ports 30303 --caplin.discovery.tcpport 40404 --caplin.discovery.port 40404 --sentinel.port 50505 --rpc.batch.limit 50000 --db.read.concurrency 8 --rpc.returndata.limit 100000000000``` 

Consensus Layer: builtin caplin

Consensus Layer Command (with flags/config): no other config specified than in the erigon commandline

Chain/Network: ethereum mainnet

#### Expected behaviour
Erigon and caplin should be able to sync up from scratch with empty datadir.

#### Actual behaviour
Everything works up to the very end, where it fails with this:

```
[INFO] [12-23|12:16:11.849] [Caplin] Forward Sync                    app=caplin stage=ForwardSync progress=10677273 distance-from-chain-tip=8m12s estimated-time-remaining=999h0m0s                                                            
[WARN] [12-23|12:16:14.075] [Caplin] Failed to process block batch   app=caplin stage=ForwardSync err="bad blocks segment received: AddChainSegment: processBlock: failed to process block body operations: outstanding deposits do not match m
aximum deposits, parentRoot; 83931c1467c16826a4f50740b03bbd00abe0eb0e3f808a907393d438577a0599"                                                  
```

The last WARN is spammed into the logs, and the sync never finishes, on the caplin RPC port, caplin reports that it is not synced.

#### Steps to reproduce the behaviour

Download alpha7, create new empty datadir, start erigon with the provided options.

#### Backtrace

No crash.

**Comment by Giulio2002** [comment]
try restarting, i saw this once already

**Comment by errge** [comment]
Yes, interestingly restart now helped.  I asked for help, because I already tried a restart, and then it happened a second time 60 minutes later at a later point in sync, but after second restart, the stuff is now fully synced and I checked queries to caplin, and all answers seems to be correct at first glance (historical and current ones too).

So, I needed to do the restart procedure twice during a sync, and I was just not persistent enough, but now I'm all synced, thank you!
